orchestra/cmds/update.py

```diff
@@ -96,20 +95,24 @@ def pull_binary_archive(name, config):
 def clone_binary_archive(name, url, config):
     logger.info(f"Trying to clone binary archive from remote {name} ({url})")
     binary_archive_path = os.path.join(config.binary_archives_dir, name)
-    env = dict(os.environ)
+    env = os.environ.copy()
     env["GIT_SSH_COMMAND"] = "ssh -oControlPath=~/.ssh/ssh-mux-%r@%h:%p -oControlMaster=auto -o ControlPersist=10"
     env["GIT_LFS_SKIP_SMUDGE"] = "1"
-    result = subprocess.run(["git", "clone", url, binary_archive_path], env=env)
+    result = run_subprocess(
+        ["git", "clone", url, binary_archive_path],
+        environment=env,
+        loglevel="DEBUG",
+    )
     if result.returncode:
         logger.info(f"Could not clone binary archive from remote {name}!")
 
 
 def git_pull(directory):
-    env = os.environ
+    env = os.environ.copy()
     env["GIT_LFS_SKIP_SMUDGE"] = "1"
-    result = subprocess.run(["git", "-C", directory, "pull", "--ff-only"],
-                            env=env,
-                            stdout=subprocess.PIPE,
-                            stderr=subprocess.STDOUT)
-    logger.info(result.stdout.decode("utf8").strip())
+    result = run_subprocess(
```

Comment:
`run_subprocess_internal`?
`loglevel="DEBUG"` should be in a single place, it's not supposed to be a caller's choice.